src/header/extended_header.rs

```diff
@@ -41,7 +41,7 @@ impl HeaderFields for ExtendedHeader {
 
     fn size(&self) -> usize {
         ExtendedHeader::STATIC_HEADER_LENGHT_BYTE
-            + get_nof_non_zero_bytes(self.key_id).max(1) as usize
+            + min_len_in_bytes(self.key_id) as usize
```

Comment:
can we make `min_len_in_bytes()` return a `usize` already?

Reply:
Unfortunately, we need it also as u8 when setting the bitfield? Maybe a new tuple type which can be interpreted as both u8 and usize could be the solution?

Reply:
naaa, its fine this way then